### PR TITLE
#708: Pass subphase timings through to ProcStats, and write them out

### PR DESCRIPTION
**Pull request by PhilMiller**
Fixes: #708

### CODE REVIEW

**Comment by PhilMiller** [comment]
I think we should close #708 with this, since further work on models and/or strategies that make effective use of the subphase timing data is a pretty open-ended project.

**Comment by PhilMiller** [comment]
@ppebay Have a look at the format in which I'm writing out the subphase times. Will that pose a problem for the analysis tools?

File: src/vt/vrt/collection/balance/proc_stats.cc [hunk 1]
```diff
@@ -202,16 +208,29 @@ void ProcStats::outputStatsFile() {
...
+
+      auto obj_str = fmt::format("{},{},{},{},[", i, id, time, subphases);
+      for (size_t s = 0; s < subphases; s++) {
+        obj_str += std::to_string(subphase_times[s]);
+        if (s != subphases - 1)
+          obj_str += ",";
+      }
+
+      obj_str += "]\n";
+
```
Comment (PhilMiller) [inline src/vt/vrt/collection/balance/proc_stats.cc:230]: @ppebay Here's where I've modified the output

**Comment by lifflander** [comment]
@PhilMiller What happened here? Why is MPI intercept code on this branch?

Can you write a Doxygen comment documenting the new format (and current one also)?

**Comment by PhilMiller** [comment]
Rebased to clear up the integration issues.

Wrote up a full description of the stats file output in the comments.

**Comment by lifflander** [comment]
This seems to be failing Travis CI sometimes

**Pull request merged**